### ISSUE TITLE
Allow users to customize date format in default name of Tab Stash groups

### ISSUE THREAD

**Issue by alxpettit** [josh-berry/tab-stash#147]
### Problem Statement

When Tab Stash makes a tab group, by default, it seems to be hardcoded to use the American date format (MM/DD/YYYY).

Ideally, date/format should be settable to whatever the end-user wants. I, for instance, would like to use the ISO8601 date/time format (YYYY-MM-DD) -- as I do with every other program on my machine. Having inconsistent date/time formats makes it harder and more confusing, when interpreting the time.

### Preferred Solution(s)

Have an option to set the time format as a [strftime](https://strftime.org/) string, or similar.

### Alternative Solution(s)

Respect browser locale settings.

### Additional Context

<hr>
<em>Readers: If you are also interested in seeing this feature be developed, please vote for it by giving the ORIGINAL POST a thumbs-up using the :smiley: button.  You are welcome to leave comments and discuss the feature request, but "Me too!" comments are not counted by the voting system.</em>

**Comment by josh-berry** [comment]
Hm, so Tab Stash should already be respecting your browser's locale settings—it actually saves bookmark folders with dates in the ISO format and translates them. (And I know it at least respects language, because I have seen screenshots from other languages where the native date/time was used as expected.)

I did a little digging, and it seems likely this is caused by a years-old Firefox bug where JS APIs are not correctly picking up on locale settings from the OS: https://bugzilla.mozilla.org/show_bug.cgi?id=476419

So I'll keep the custom date format idea in mind, but unfortunately I think we're probably waiting on Mozilla for a fix for the actual issue. :/

**Comment by alxpettit** [comment]
Damn! Yeah, you're right, I'm on Firefox. :P Whoops!

Honestly, I think it should be possible to override locale settings and set a custom date regardless -- ideally. It's not unusual for someone to live in one place but dislike the dictates of the locale there. I actually ended up fabricating a custom locale that functions like en-DK just because I live in the United States, but the locale formats for here are frankly insane IMO.

Unfortunately, there are a lot of cases where it doesn't work without weird and complicated hacks -- for instance, I have a script that copies it into Fatpak containers, otherwise it gets ignored by most Flatpak apps. Honestly I think the best (as in, most portable) standard, given the world we're in today, is to have it possible to set whatever date format you want with strftime or another format.

Unfortunately, Javascript has some pretty limited core libraries relative to other scripting languages... so unless you feel like adding a Moment.js dependency, I'm not sure what to suggest. :P

**Comment by josh-berry** [comment]
Yeah I'm with you there (I'm also in the US, though my pet peeves are 12-hour time and Fahrenheit).

I know there is a JS proposal to improve the whole date/time situation in the core libraries, but that might take some time to come to fruition. I'll keep an eye on it though.

**Comment by lasers** [comment]
I'd like this feature too...

Change default name so `Saved 6/13/2023` could be replaced with `Stashed 6/13/2023`.... or really, to just add brackets `[Saved MM/DD/YYYY]` to default name so the bookmark-sorting extension would keep the unnamed stashed groups at top.

Change date format can allow users to be a bit more flexible by stashing various tabs (not grouped) to new or existing current day/week/month group (no timestamp).
